File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let convert = 0,
        erase = 0;

    while (s !== "1") {
        erase += s.split("0").length - 1;
        s = s.replace(/0/g, "");
        ++convert;

        s = s.length.toString(2);
    }

    return [convert, erase];
}

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <bitset>

using namespace std;

vector<int> solution(string s) {
    int count{0}, erase{0};

    while (s != "1")
    {
        while (find(s.begin(), s.end(), '0') != s.end())
        {
            s.erase(find(s.begin(), s.end(), '0'));
            ++erase;
        }

        s = bitset<100>(s.length()).to_string();
        s = s.substr(s.find('1'));
        ++count;
    }

    return vector<int>{ count, erase };
}